src/components/CookieConsent/CookieConsent.tsx

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+
+import {useMobile} from '@gravity-ui/uikit';
+
+import {block} from '../utils/cn';
+
+import {ConsentMode} from './ConsentManager';
+import {ConsentNotification} from './components/ConsentNotification/ConsentNotification';
+import {ConsentPopup} from './components/ConsentPopup/ConsentPopup';
+import {SimpleConsent} from './components/SimpleConsent/SimpleConsent';
+import {CookieConsentBaseProps, CookieConsentProps} from './types';
+
+const b = block('analytics');
+
+export const CookieConsent = ({
+    consentManager,
+    onConsentPopupClose,
+    disableInitialOpen,
+    onAction,
+    forceOpen,
+    ...popupProps
+}: CookieConsentProps) => {
+    const [isOpened, setIsOpened] = React.useState(false);
+    const [mobile] = useMobile();
+
+    const isNotificationMode = consentManager.mode === ConsentMode.Notification;
+
+    React.useEffect(() => {
+        // Show banner after some timeout so that the user has time to see the service content
+        const timeoutId = setTimeout(() => {
+            const isConsentsDefined = consentManager.isAllConsentsDefined();
+            const shouldOpen = isNotificationMode
+                ? !consentManager.isClosed() || !isConsentsDefined
+                : !isConsentsDefined;
+            const differentEdition = consentManager.projectEdition !== consentManager.edition;
```

Comment:
Seems, this logic should be inside `consentManager`

Reply:
Disagree, consentManager works only with cookies and managing them
Logic about component visibility should be inside the component